File: src/main/java/com/classroom/user/service/impl/SecurityServiceImpl.java
package com.classroom.user.service.impl;

import com.classroom.user.service.SecurityService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.stereotype.Service;

@Service
@SuppressWarnings("unused")
public class SecurityServiceImpl implements SecurityService {

    @Autowired
    private UserDetailsService userDetailsService;

    @Autowired
    private AuthenticationManager authenticationManager;

    private static final Logger logger = LoggerFactory.getLogger(SecurityServiceImpl.class);

    @Override
    public String findLoggedInUsername() {
        var userDetails = SecurityContextHolder.getContext().getAuthentication().getDetails();
        if (userDetails instanceof UserDetails details) {
            return details.getUsername();
        }
        return null;
    }

    @Override
    public void autologin(final String username, final String password) {
        logger.info("Getting login for user : {} with pass : {}", username, password);
        var userDetails = userDetailsService.loadUserByUsername(username);
        var token =
                new UsernamePasswordAuthenticationToken(userDetails, password, userDetails.getAuthorities());
        try {
            authenticationManager.authenticate(token);
        } catch (Exception ex) {
            logger.error(ex.getMessage());
        }
        if (token.isAuthenticated()) {
            SecurityContextHolder.getContext().setAuthentication(token);
            logger.debug("Auto login {} successfully!", username);
        }
    }
}

File: src/main/kotlin/com/classroom/user/service/impl/SecurityServiceImpl.kt
package com.classroom.user.service.impl

import com.classroom.user.service.SecurityService
import org.slf4j.LoggerFactory
import org.springframework.security.authentication.AnonymousAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.stereotype.Service

@Service
class SecurityServiceImpl : SecurityService {

    private val logger = LoggerFactory.getLogger(javaClass)

    override fun findLoggedInUsername(): String? =
        (SecurityContextHolder.getContext().authentication?.details as? UserDetails)?.username

    fun isAuthenticated(): Boolean {
        val authentication = SecurityContextHolder.getContext().authentication
        return authentication != null &&
                authentication !is AnonymousAuthenticationToken &&
                authentication.isAuthenticated
    }
}